dataplane/Dockerfile

```diff
@@ -1,26 +1,42 @@
-FROM archlinux as builder
+FROM ubuntu:23.10 as builder
 
-RUN pacman -Syu --noconfirm
-RUN pacman -S base-devel protobuf rustup --noconfirm
+ARG TARGETARCH
 
+RUN apt update
+RUN apt install --yes build-essential protobuf-compiler curl pkg-config llvm
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
```

Comment:
I did like with Arch that `rustup` was one of the signed packages in the default repositories and not just a `curl` into root `bash` line...

Looking back on this change, it seems the goal was to enable cross-compile, but we switched to Ubuntu it would seem to enable building ON non x86 machines... do you think it would be reasonable to stick with Arch as the builder and just enable cross-compilation?

Reply:
Talked about this in Slack: we can't simply enable cross-compilation, apparently there is a need to build able to build on ARM systems.

So @aryan9600 I think I have a solution: Perhaps we can just use the [Official Rust Image](https://hub.docker.com/_/rust/). This comes with `rustup` by default and is based on Debian, plus it's multi-arch and available on ARM? Can't remember the reason we used Arch as the builder in the past, but I imagine it had something to do with the protocol buffers support, I think we should see if we can get all that working on the `rust` image, LMKWYT? :thinking:

Reply:
sounds good to me!